Patch-Source: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4567
From 7dd89421bfd31f851a0aa6379ee148bd0f688d4d Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 14 Mar 2022 08:44:09 -0400
Subject: [PATCH] build: Avoid objcopy on arm

The trickery we do with objcopy and ld to speed up
resource inclusion does not seem to work right on
32bit Arm, so just skip it there.

Fixes: #4757, #4748, #4752
---
 demos/gtk-demo/meson.build       | 2 +-
 demos/widget-factory/meson.build | 2 +-
 gtk/meson.build                  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/demos/gtk-demo/meson.build b/demos/gtk-demo/meson.build
index f594cd59b2..1347baf2aa 100644
--- a/demos/gtk-demo/meson.build
+++ b/demos/gtk-demo/meson.build
@@ -160,7 +160,7 @@ endif
 
 ld = find_program('ld', required : false)
 
-if build_machine.system() == 'linux' and objcopy.found() and objcopy_supports_add_symbol and ld.found()
+if not meson.is_cross_build() and build_machine.cpu_family() != 'arm' and build_machine.system() == 'linux' and objcopy.found() and objcopy_supports_add_symbol and ld.found()
   glib_compile_resources = find_program('glib-compile-resources')
 
   # Create the resource blob
diff --git a/demos/widget-factory/meson.build b/demos/widget-factory/meson.build
index 9a083eb1ac..550eb09bd9 100644
--- a/demos/widget-factory/meson.build
+++ b/demos/widget-factory/meson.build
@@ -8,7 +8,7 @@ endif
 
 ld = find_program('ld', required : false)
 
-if build_machine.system() == 'linux' and objcopy.found() and objcopy_supports_add_symbol and ld.found()
+if not meson.is_cross_build() and build_machine.cpu_family() != 'arm' and build_machine.system() == 'linux' and objcopy.found() and objcopy_supports_add_symbol and ld.found()
   glib_compile_resources = find_program('glib-compile-resources')
 
   # Create the resource blob
diff --git a/gtk/meson.build b/gtk/meson.build
index 9e01201522..3b77936e3c 100644
--- a/gtk/meson.build
+++ b/gtk/meson.build
@@ -872,7 +872,7 @@ endif
 
 ld = find_program('ld', required : false)
 
-if not meson.is_cross_build() and build_machine.system() == 'linux' and objcopy.found() and objcopy_supports_add_symbol and ld.found()
+if not meson.is_cross_build() and build_machine.cpu_family() != 'arm' and build_machine.system() == 'linux' and objcopy.found() and objcopy_supports_add_symbol and ld.found()
   glib_compile_resources = find_program('glib-compile-resources')
 
   # Create the resource blob
-- 
2.35.1

